### PR TITLE
Task 7 add customers

### PR DESCRIPTION
**Pull request by 0thiago**

### CODE REVIEW

File: Assets/Scripts/Customers/Customer.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Customer : MonoBehaviour
+{
+    [SerializeField] private float _walkSpeed;
```
Comment (Leo-gp) [inline Assets/Scripts/Customers/Customer.cs]: Can we put `walkSpeed` under a scriptable object named `CustomerConfiguration`? You can refer to the SO here and access the `walkSpeed`.

File: Assets/Scripts/Customers/Customer.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    [SerializeField] private float _despawnCustomerPointOnX = 11f;
```
Comment (Leo-gp) [inline Assets/Scripts/Customers/Customer.cs]: Can we not prefix the SerializeField ones with underscore? Conventional non-serialized private fields can have the underscore though.

File: Assets/Scripts/Managers/CustomerSpawnManager.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CustomerSpawnManager : MonoBehaviour
```
Comment (Leo-gp) [inline Assets/Scripts/Managers/CustomerSpawnManager.cs:5]: Can we put `spawnFrequencyTime` and `spawnCustomerHeight` under a scriptable object named `CustomerConfiguration`? You can refer to the SO here in this class and access these fields.

File: Assets/Scripts/Customers/Customer.cs (hunk 2 above)
Comment (0thiago) [reply Assets/Scripts/Customers/Customer.cs]: - serialized private and local var without underscore.
- non-serialized private with underscore.
- public without underscore and stating with Captital letter.
right?

File: Assets/Graphics/Materials/Black.mat
Comment (Leo-gp) [inline Assets/Graphics/Materials/Black.mat:1]: We don't need to use materials to color the sprites. The SpriteRenderer component itself has a color property. You can change it there directly. However, you can just keep all of them uncolored, as we'll have their sprites coming in with their colors already.
TLDR: Remove the material assets and keep all sprites as white squares for now.

File: Assets/Scripts/Customers/Customer.cs (hunk 2 above)
Comment (Leo-gp) [reply Assets/Scripts/Customers/Customer.cs]: Yes right. Public ones being necessarily properties.
Example: 
```c#
public int WalkSpeed { get; private set; }
```
Above example is what you'd do to access the SO field on the `Customer` btw.

Comment (0thiago) [reply Assets/Scripts/Customers/Customer.cs]: Not sure what you mean, we create on SO this line `public int WalkSpeed { get; private set; }` in order to be able to access  WalkSpeed on Customers script?

Comment (Leo-gp) [reply Assets/Scripts/Customers/Customer.cs]: Sorry, above property example would apply to fields you'd want to set internally in the class and access outside in other classes. For the SO, since we set the values from Unity editor, you'd have a serialized field to set in the SO inspector:
```c#
[SerializeField] private float walkSpeed;
```
and a public property for other classes to read it:
```c#
public float WalkSpeed => walkSpeed;
```

File: Assets/Scripts/Managers/CustomerSpawnManager.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+{
+    [SerializeField] private GameObject[] customersList;
```
Comment (Leo-gp) [inline Assets/Scripts/Managers/CustomerSpawnManager.cs]: Can refer to the `Customer` class to avoid attaching some wrong game object.

File: Assets/Scripts/Managers/CustomerSpawnManager.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    private void Start()
+    {
+        InvokeRepeating(nameof(SpawnCustomer), customerConfiguration.SpawnFrequencyTime, customerConfiguration.SpawnFrequencyTime);    
+    }
+
+    private void SpawnCustomer()
+    {
+        int attempts = 0; // Avoid infinty loop
+
+        while (attempts < customersList.Length)
```
Comment (Leo-gp) [inline Assets/Scripts/Managers/CustomerSpawnManager.cs]: Can we use a for loop instead of while? Can iterate `customersList` until the first inactive one is found. No need to pick random ones.

File: Assets/Scripts/Managers/CustomerSpawnManager.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            int randomIndex = Random.Range(0, customersList.Length);
+            GameObject customer = customersList[randomIndex];
+
+            if (!customer.activeInHierarchy)
+            {
+                customer.SetActive(true);
+                customer.transform.localPosition = new(customer.transform.localPosition.x, customerConfiguration.SpawnCustomerHeightPositionOnScreen);
+
+                Debug.Log(customer.gameObject + " - transform: " + customer.transform.position);
```
Comment (Leo-gp) [inline Assets/Scripts/Managers/CustomerSpawnManager.cs]: Can delete Debug.Log's after local debugging. They have performance hit if we keep them.

**Review by Leo-gp — CHANGES_REQUESTED** [review]

**Review by Leo-gp — APPROVED** [review]
